### DISCUSSION TITLE
Can't figure out how to specify System.DateTime.TryParse method in Trimmer descriptor XML file to have it retain this method overload

### DISCUSSION THREAD

**Discussion by rkeithhill-keysight** [dotnet/runtime#109471] [Q&A]
We have a .NET library that dynamically loads a C++/CLI assembly from another app's install dir. I'm experimenting with using a Root descriptor XML file to tell the trimmer for my sample app, not to trim this a particular overload.  The overload is:
```text
System.MissingMethodException: Method not found: 'Boolean System.DateTime.TryParse(System.String, System.IFormatProvider, System.Globalization.DateTimeStyles, System.DateTime ByRef)'
```
I have tried variations specifying this method (or the entire DateTime type):
```xml
<linker>
  <assembly fullname="System.Private.CoreLib">
    <type fullname="System.Convert">
      <method signature="System.String ToString(System.Boolean,System.IFormatProvider)"/>
    </type>
    <!-- <type fullname="System.DateTime"/> -->
    <!-- public static bool TryParse([NotNullWhen(true)] string? s, IFormatProvider? provider, DateTimeStyles styles, out DateTime result) -->
      <!-- <method signature="System.Boolean TryParse(System.String,System.IFormatProvider,System.Globalization.DateTimeStyles,System.DateTime ByRef)"/> -->
      <!-- <method name="TryParse"/> -->
    <!-- </type> -->
  </assembly>
  <assembly fullname="System.Runtime">
    <type fullname="System.AppDomain">
      <method name="IsDefaultAppDomain"/>
    </type>
    <type fullname="System.DateTime"/>
```
I think DateTime is actually implemented/provided in `System.Private.CoreLib.dll` even though the API docs say `System.Runtime.dll`. 

I've tried the simple `<type fullname="System.DateTime"/>` to keep all of DateTime.  And I've tried specifying the `signature`.  BTW it's not obvious in the [docs ](https://github.com/dotnet/runtime/blob/main/docs/tools/illink/data-formats.md#descriptor-format) how to specify a signature that has an `out` parameter.  Also, the signature format does not seem to be forgiving RE extra spaces - so, no spaces between parameters.

The only way I've gotten my sample/test app to get past this missing method exception is to add a call to this DateTime.TryParse method overload in the app.  That's fine for my testing, but not for folks who use my library.  At the end of the day, I figure it won't be realistic to provide my library as "trim-enabled", but I'd like to get a bit further to see how big the problem is.

**Comment by huoyaoyuan** [comment]
Did you inspect the trimmed assemblies with ILSpy, to verify which are present/missing in `System.Runtime` and `System.Private.CoreLib`?

**Reply by rkeithhill-keysight** [reply]
I used dotPeek and found that `System.Runtime.dll` type forwards DateTime to `System.Private.CoreLib.dll` where the DateTime.TryParse overload is actually implemented:

![image](https://github.com/user-attachments/assets/1719a5cd-eed0-4631-8cf6-16e08d611ea5)

The trimmed version does not have this particular overload. That's why I'm trying to figure out how to get the trimmer to not trim it.  I can't seem to find an XML syntax/format that works.
